packages/ui/src/views/layouts/SandboxLayout.tsx

```diff
@@ -88,32 +88,36 @@ function Main({
   hasSubHeader?: boolean
   hasLeftPanel?: boolean
   hasLeftSubPanel?: boolean
+  leftSubPanelWidth?: number
 }) {
   const paddingTopClass =
     hasHeader && hasSubHeader ? `pt-[calc(55px+45px)]` : hasHeader ? 'pt-[55px]' : hasSubHeader ? 'pt-[45px]' : ''
 
-  const paddingLeftClass =
-    hasLeftPanel && hasLeftSubPanel
-      ? 'pl-[calc(220px+300px)]'
-      : hasLeftPanel
-        ? 'pl-[220px]'
-        : hasLeftSubPanel
-          ? 'pl-[300px]'
-          : ''
+  const paddingLeftStyle = {
+    paddingLeft:
+      hasLeftPanel && hasLeftSubPanel
+        ? `calc(220px + ${leftSubPanelWidth}px)`
+        : hasLeftPanel
+          ? '220px'
+          : hasLeftSubPanel
+            ? `${leftSubPanelWidth}px`
+            : undefined
+  } as CSSProperties
```

Comment:
Instead of overriding the type with `as`, set the variable type. `as` tells TS to ignore what it knows and trust you; setting the variable type tells TS to make sure you're correct.
```suggestion
  const paddingLeftStyle: CSSProperties = {
    paddingLeft:
      hasLeftPanel && hasLeftSubPanel
        ? `calc(220px + ${leftSubPanelWidth}px)`
        : hasLeftPanel
          ? '220px'
          : hasLeftSubPanel
            ? `${leftSubPanelWidth}px`
            : undefined
  }
```